src/main/java/org/elasticsearch/cluster/routing/UnassignedInfo.java

```diff
@@ -190,7 +190,7 @@ public static int getNumberOfDelayedUnassigned(Settings settings, ClusterState s
             if (shard.primary() == false) {
                 IndexMetaData indexMetaData = state.metaData().index(shard.getIndex());
                 long delay = shard.unassignedInfo().getDelayAllocationExpirationIn(settings, indexMetaData.getSettings());
-                if (delay > 0) {
+                if (delay != 0) {
```

Comment:
Ah! I get it! 0 means they are not delayed at all but a negative number means they've already expired. They _aren't_ delayed any more. OK - I retract my question. Maybe a comment?